geomesa-gs-wfs/src/main/scala/org/geomesa/gs/wfs/output/ArrowOutputFormat.scala

```diff
@@ -76,40 +80,24 @@ class ArrowOutputFormat(geoServer: GeoServer)
             case _ => false
           }
-          if (aggregated) {
-            // with distributed processing, encodings have already been computed in the servers
-            iter.map(_.getAttribute(0).asInstanceOf[Array[Byte]]).foreach(bos.write)
-          } else {
-            // for non-encoded fs we do the encoding here
-            logger.warn(s"Server side arrow aggregation is not enabled for feature collection '${fc.getClass}'")
-            val request = GetFeatureRequest.adapt(getFeature.getParameters()(0))
-            val hints = new Hints()
-            populateFormatOptions(request, hints)
-
-            val encoding = SimpleFeatureEncoding.min(hints.isArrowIncludeFid, hints.isArrowProxyFid, hints.isFlipAxisOrder)
-            val dictionaries = hints.getArrowDictionaryFields
-            val version = hints.getArrowFormatVersion.getOrElse(FormatVersion.ArrowFormatVersion.get)
-            val sortField = hints.getArrowSort.map(_._1)
-            val sortReverse = hints.getArrowSort.map(_._2)
-            val batchSize = hints.getArrowBatchSize.getOrElse(ArrowProperties.BatchSize.get.toInt)
-            val flattenStruct = hints.isArrowFlatten
-
-            val preSorted = for (field <- sortField; reverse <- sortReverse.orElse(Some(false))) yield {
-              request.getQueries.get(i).getSortBy match {
-                case list if list.size == 1 =>
-                  val sort = list.get(0)
-                  Option(sort.getPropertyName).exists(_.getPropertyName == field) &&
-                      (sort.getSortOrder == SortOrder.DESCENDING) == reverse
-                case _ => false
+          hints match {
+            case _ if !aggregated =>
+              // for non-encoded fs we do the encoding here
+              logger.warn(s"Server side arrow aggregation not enabled for feature collection: ${fc.getClass}, SFT type: ${featureType.getTypeName}")
+              arrowVisitor(request, hints, i, bos, iter, featureType)
+            case _ if !hints.isArrowFlatten => // && aggregated
+              // with distributed processing, encodings have already been computed in the servers
+              iter.map(_.getAttribute(0).asInstanceOf[Array[Byte]]).foreach(bos.write)
+            case _ if !hints.isArrowProcessDeltas => // && aggregated && arrowFlatten
+              // not able to read the byte arrays with a SimpleFeatureArrowFileReader because they're a custom format
+              // that includes threading keys and dictionary deltas
+              iter.map(_.getAttribute(0).asInstanceOf[Array[Byte]]).foreach(bos.write)
+            case _ => // aggregated && arrowFlatten && processDeltas
+              WithClose(SimpleFeatureArrowFileReader.streaming(() => featureSequenceInputStream(iter))) { reader =>
+                WithClose(reader.features()) { features =>
+                  arrowVisitor(request, hints, i, bos, features, reader.sft)
+                }
               }
-            }
```

Comment:
i think this would be clearer as just a regular if statement, since we're not matching anything (everything is `case _`). but the logic should be:

```
if (!aggregated) {
   arrowVisitor(...)
} else if (hints.isArrowFlatten) {
  if (!hints.isArrowProcessDeltas) {
    throw new NotImplementedError("Can't flatten + not process deltas")
  }
  WithClose(...)
} else {
  iter.map(_.getAttribute(0).asInstanceOf[Array[Byte]]).foreach(bos.write)
}
```